### PR TITLE
Feature: Frame Overlays

### LINKED ISSUE #61: Frame Overlays
## Feature Request

**Description:**
Visually present key information about frames on a given page. A semi-transparent overlay would be rendered on top of frames on the page, and within the overlay, key information about the frame is presented. 

**User Story:**
As a developer/user, I want understand the status and behaviors of embedded content on a given page and the corresponding cookies, so that I can easily assess the impact of blocked 3P cookies on the different parts of a site.

**Acceptance Criteria:**
- [ ] Frame identification
- [ ] Frame overlay rendering
- [ ] Display frame information (e.g. permission policies, cookies)

cc: @kevinkiklee, @sayedtaqui, @pradeep910

### PR DESCRIPTION
**Pull request by sayedtaqui**
## Description

Allow developers to inspect cookies and other information associated with iframes from the “Privacy Sandbox” panel in the developer tool.

## Relevant Technical Choices

Chrome [message passing API](https://developer.chrome.com/docs/extensions/mv3/messaging/) was used to communicate between the Developer tool and the content script on the web page as shown below in the diagram

image

## Testing Instructions

Use https://psanalysis.site for testing all frames

### Limitations

- **Edge Cases**
    - One case was found on http://bbc.com/) where a link overlay was covering the iframe. The tool was unable to detect the iframe by hovering over it because the iframe is behind the link. (iframe src: //emp.bbc.com/emp/SMPj/2.50.8/iframe.html)
    - There are some cases where iframes are nested inside another iframes, the tool is unable to detect those frames. Example https://ads.stickyadstv.com/ and https://cm.adgrx.com/ on bbc.com
    
- **Know Issues**
    - Sometimes, when scrolling quickly and trying to hover over an iframe at the same time, the tool may fail to detect the iframe and won't display the overlay. However, it will appear once you move out of the iframe and return to it.
    - Occasionally, while hovering over an iframe, if you suddenly move your cursor outside of the tab, the inspected iframe tooltip will be scrolled into the view.

## Screenshot/Screencast

<kbd>
image
</kbd>

Fixes #61

### CODE REVIEW

**Comment by sayedtaqui** [comment]
@mayan-000 I did a little refactoring of the RWS code so that it can be reused in the tool, please review that part.

**Review by mayan-000 — COMMENTED** [review]
RWS code LGTM ✅

File: packages/design-system/src/components/errorFallback/index.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+const ErrorFallback = ({ error }: ErrorFallbackProps) => {
+  // Call resetErrorBoundary() to reset the error boundary and retry the render.
+
+  return (
+    <div className="p-4 text-center" role="alert">
+      <p className="font-bold mb-5 text-xl">Something went wrong!</p>
+      <p className="mb-10">Please try opening this page in a new tab.</p>
+      <pre className="text-red-700 text-xs bg-slate-50 p-5 text-left border-l-4">
+        {error?.message} Please try opening this page in a new tab. Please try
```
Comment (ayushnirwal) [inline packages/design-system/src/components/errorFallback/index.tsx:33]: Sentence is repeated.

**Review by ayushnirwal — APPROVED** [review]
LGTM

**Review by amovar18 — APPROVED** [review]
LGTM!

**Pull request merged**